### PR TITLE
Feature/user additional location

### PR DESCRIPTION
**Pull request by IvanUglovets**
added multiple choice of locations
refactored the code
image
image
image
image

### CODE REVIEW

**Review by x8er — APPROVED** [review]

File: packages/apiserver/src/migrations/1661339543400-migrationMy.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  name = 'migrationMy1661339543400';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`ALTER TABLE "person" DROP COLUMN "location_id"`);
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`ALTER TABLE "person" ADD "location_id" integer`);
+  }
+}
```
Comment (unrealsolver) [inline packages/apiserver/src/migrations/1661339543400-migrationMy.ts:13]: Are there any reasons not to merge these migrations into one?
Also, could you use more sane name other than `migrationMy` please?

**Review by unrealsolver — CHANGES_REQUESTED** [review]

File: packages/frontapp/src/app/components/DonateBook/index.tsx [hunk 2]
```diff
@@ -233,14 +240,16 @@ interface IPropsDonateBook {
   onHideContent: () => void;
 }
 const DonateBook: FC<IPropsDonateBook> = ({ data, onHideContent }) => {
-  const { id, location, userRole } = useAppSelector((state) => state.user);
+  const { id, userRole } = useAppSelector((state) => state.user);
   const { identifier } = useAppSelector((state) => state.identifier);
+  const { locations } = useAppSelector((state) => state.user);
```
Comment (lexussuxel) [inline packages/frontapp/src/app/components/DonateBook/index.tsx:245]: Why you use useAppSelector for user two times?

File: packages/frontapp/src/app/components/DropdownLocation/index.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+      background-color: ${colors.accent_color};
+      border: ${dimensions.base} solid transparent;
+      background-clip: padding-box;
+    }
+  }
+`;
+
+const CheckBox = styled.input`
+  width: 24px;
+  height: 24px;
```
Comment (lexussuxel) [inline packages/frontapp/src/app/components/DropdownLocation/index.tsx:104]: you can use fixed value xl_2

File: packages/frontapp/src/app/components/OneDonatedBookPreview/index.tsx [hunk 4]
```diff
@@ -131,6 +131,8 @@ const OneDonator = ({
     navigate(`/donate/${item_id}`);
   };
   const lastStatus = statuses.slice(-1)[0];
+  console.log(statuses);
+  console.log(lastStatus);
```
Comment (lexussuxel) [inline packages/frontapp/src/app/components/OneDonatedBookPreview/index.tsx:135]: Remove this console.logs

**Review by x8er — APPROVED** [review]

File: packages/apiserver/src/app/app.module.ts [hunk 5]
```diff
@@ -81,9 +81,9 @@ import { GrantGuard } from '../permission/grant.guard';
   providers: [
     AppService,
     AppResolver,
-    { provide: APP_GUARD, useClass: AuthGuard },
-    { provide: APP_GUARD, useClass: BlockedUsersGuard },
-    { provide: APP_GUARD, useClass: GrantGuard },
+    // { provide: APP_GUARD, useClass: AuthGuard },
+    // { provide: APP_GUARD, useClass: BlockedUsersGuard },
+    // { provide: APP_GUARD, useClass: GrantGuard },
```
Comment (x8er) [inline packages/apiserver/src/app/app.module.ts:86]: why?

**Review by x8er — CHANGES_REQUESTED** [review]

**Review by x8er — APPROVED** [review]

**Review by lexussuxel — APPROVED** [review]

**Pull request merged**